ocaml/typing/jkind.mli

```diff
@@ -115,64 +152,126 @@ end
 (******************************)
 (* constants *)
 
-(** Constant jkinds are used both for user-written annotations and within
-    the type checker when we know a jkind has no variables *)
-type const = Jkind_types.const =
-  | Any
-  | Value
-  | Void
-  | Immediate64
-  | Immediate
-  | Float64
-  | Float32
-  | Word
-  | Bits32
-  | Bits64
+module Const : sig
+  (** Constant jkinds are used for user-written annotations *)
+  type t = Types.type_expr Jkind_types.Const.t
 
-val const_of_user_written_annotation :
-  context:annotation_context -> Jane_syntax.Jkind.annotation -> const
+  val to_out_jkind_const : t -> Outcometree.out_jkind_const
```

Comment:
That was the reasoning for it. Well, that's a bit of a lie. `Printtyp` would need to see inside `Jkind.Const.t` to be able to print jkinds. This is something that's not supposed to be exposed, but in reality is because of `jkind_types.ml`.